app/vulncode_app/models.py

```diff
@@ -50,13 +50,16 @@ class VulnerabilityReference(models.Model):
     class Meta:
         unique_together = ('vulnerability', 'source', 'reference_id')
```

Comment:
And you may want to check other models that may need updated unique together too